TYPING_GUIDELINES.md

```diff
@@ -0,0 +1,61 @@
+# Guidelines When adding types 
+
+When you have scenarios like: 
+```tsx
+const C = Card as typeof Card & CardComponents;
+
+C.Fit = CardFit;
+
+export * from "./Card";
+
+export default C;
+```
```

Comment:
Instead of this code, which doesn't exist anymore in our codebase, I think we should describe the problem and just show the correct solution below.

For example:

If a component has subcomponents, use `Object.assign`:

```
code
```